### ISSUE TITLE
pihole custom service - Access-Control-Allow-Credentials problem - CORS set in pihole

### ISSUE THREAD

**Issue by D4rkiii** [bastienwirtz/homer#297]
Hello,

I'm facing a problem with the pihole custom service and im not sure if i'm doing something wrong.

Setup:
Homer: dashboard.mydomain.com
config.yml:
      - name: "PiHole"
        logo: "assets/tools/sample2.png"
        #subtitle: "Network-wide Ad Blocking"
        <<: *Apps
        type: "PiHole"
        url: "http://pihole.mydomain.de/admin"

pihole: pihole.mydomain.com
CORS_HOSTS: dashboard.mydomain.com
Set the CORS_HOSTS as mentioned in this pull request from pihole or in the current documentation. (PR: https://github.com/pi-hole/docker-pi-hole/pull/867)

Before i set the CORS HOSTS in pihole i got the error message which is decribed here under the Troubleshooting section.
After i set the CORS_HOSTS im getting the following error:

**Access to fetch at 'http://pihole.mydomain.de/admin/api.php' from origin 'http://dashboard.mydomain.de' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Credentials' header in the response is '' which must be 'true' when the request's credentials mode is 'include'.**

Its because the the fetch function in the pihole.vue calls the endpoint with the credentials: "include" options.

Is there a way to disable this or do i have to configure something to get this ready?

/Edit:
@danzkigg mentioned it in the last issue about pihole stats arent showing up. Issue: #194 

Best regards
Tobi

**Comment by AlexFullmoon** [comment]
Don't know much about how how pihole does it or php handles CORS, but I'd like to note that homer modules send HEAD requests, which might not be accounted for in usual settings.

Just had to deal with nginx configuration, and most examples I googled up check only for GET, POST, OPTIONS and sometimes a few others, but not HEAD. E.g. this bit in locations block:
```nginx
if ($request_method = OPTIONS) {
    add_header Access-Control-Allow-Origin "$http_origin" always;
... }
if ($request_method ~ '(GET|POST)') {
    add_header Access-Control-Allow-Origin "$http_origin" always;
... }
```
In such case HEAD request won't have header set. Maybe builtin pihole server has similar settings?

As temporary solution you might use reverse proxy in front of pihole with something akin to nginx `proxy_hide_header   Access-Control-Allow-Origin;`

**Comment by mcclurec** [comment]
This is a duplicate of [issue 296](https://github.com/bastienwirtz/homer/issues/296). I inadvertently introduced this issue in [PR 258](https://github.com/bastienwirtz/homer/pull/258). I'll work on a fix to make the `credentials: 'include'` param optional. Ah, the CORS... always a joy to work with. Turns out if we all have different networking stacks, it's hard to make universal decision that "just work". Sorry for the trouble.

@bastienwirtz Recommend closing this as a dupe

**Comment by D4rkiii** [comment]
Ahhh sorry @mcclurec not seen this issue and it wasnt my purpose do add a duplicate, sorry for that!
Thanks for the input and for the clearification! I will close this ticket with this comment and If you need to know how to reproduce this.

Here my setup in my local network:
- pihole (docker on synology nas)
- homer (docker on synology nas)
- Both registered in reverse proxy from synology (simple ui no advanced configuration needed)
-- domains below as source in reverse proxy and target host the docker container ip
- pihole local dns entry for both webs pointing to the synology. (pihole.domain.com / homer.domain.com)
- No authentication in front of pihole & homer

**Comment by mcclurec** [comment]
No worries about the dupe! Thanks for the extra info
